### DISCUSSION TITLE
Server side filter

### DISCUSSION THREAD

**Discussion by chnamoore** [Mimetis/Dotmim.Sync#958] [Q&A]
I'm trying to figure out how to do this...

Now that I have my mobile devices synchronizing tables with my web proxy, I have the need to filter rows from a few tables because they are getting large and the data is no longer needed on the mobile device.

My idea is just to create a field called "SyncToMobile" in my database table and let users control (from a PC app) which records sync with the mobile device by setting a field SyncToMobile=true.

For instance lets say I have a table "Orders" and a related table "OrderLines". Can I add a column "SyncToMobile" to the "Orders" table and have the web proxy sync just the Order rows where SyncToMobile=true and the related OrderLines rows to the mobile devices?

1.) How do I go about doing this?
2.) Also, do I have to do anything special to remove the old records from the mobile devices when the SyncToMobile flag eventually is set to false?

Aaron

**Comment by chnamoore** [comment]
Nobody knows how I can achieve this?

**Comment by ethhill** [comment]
We have a similar need, but with soft deletes.  On the initial sync we don't want any rows where the Deleted column is true, but on subsequent sync's we want the soft deleted rows sync'd soi that the rows get marked deleted on the client.

We accomplished this by intercepting OnRowsChangesSelected, checking if args.Context.SyncType is reinitialize, and if args.SyncRow["Deleted"] we set SyncRow to null.  This keeps the soft deleted rows from initially sync'ing but includes updated soft deleted rows on subsequent sync operations.

**Comment by christoneethling** [comment]
Hi @chnamoore, 
Not sure if the solution suggested by @ethhill was good enough? 
I create a sample app that shows this approach and some variations on it  [here](https://github.com/Mimetis/Dotmim.Sync/discussions/1367#discussion-9169285), not sure if would be of interest.
